### PR TITLE
chore: Refactor profiles usage to projects

### PR DESCRIPTION
**Pull request by onuzbee**
# Fixes #
Closes [DI-1475](https://issues.corp.twilio.com/browse/DI-1475)
As we are adding a new JSON object for profiles, the existing code using the profiles object (which is reading the projects array) is being refactored. 

Dependent PR in twilio-cli : [Link](https://github.com/twilio/twilio-cli/pull/257)
# Testing 

- Tested Unit Tests : [Test Report Link](https://gist.github.com/onuzbee/4d64268a7927b4b472e6f0c4757057dd)
- Tested profiles usage - create/remove/list manually 

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli-core/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: src/services/config.js [hunk 1]
```diff
@@ -146,9 +146,8 @@ class ConfigData {
     this.edge = configObj.edge;
     this.email = configObj.email || {};
     this.prompts = configObj.prompts || {};
-    // Note the historical 'projects' naming.
-    configObj.profiles = configObj.projects || [];
-    configObj.profiles.forEach((profile) => this.addProfile(profile.id, profile.accountSid, profile.region));
+    configObj.projects = configObj.projects || [];
+    configObj.projects.forEach((project) => this.addProfile(project.id, project.accountSid, project.region));
     this.setActiveProfile(configObj.activeProject);
```
Comment (childish-sambino) [inline src/services/config.js:151]: Removed comment still applies to this line for profile/project.

File: src/services/config.js [hunk 2]
```diff
@@ -180,8 +179,7 @@ class Config {
       edge: configData.edge,
       email: configData.email,
       prompts: configData.prompts,
-      // Note the historical 'projects' naming.
-      projects: configData.profiles,
+      projects: configData.projects,
       activeProject: configData.activeProfile,
```
Comment (childish-sambino) [inline src/services/config.js:183]: Same for this line.

**Review by childish-sambino — APPROVED** [review]
Think it's best to create a new pre-release branch and update this PR to merge to that branch. That way, all the breaking changes can go against the same branch and not affect `main` until ready.

**Comment by Sindhura3** [comment]
@onuzbee Please reparent this to release-feature-branch instead of main and change the PR too

**Comment by onuzbee** [comment]
Thanks for pointing, github rebased it back to track main with a second push. 
Manually based it back to release-feature-branch

File: src/services/config.js (hunk 1 above)
Comment (onuzbee) [reply src/services/config.js:152]: Added back comments at both locations.

**Review by ravali-rimmalapudi — APPROVED** [review]
Looks Good.

**Pull request merged**